src/ray/util/util.h

```diff
@@ -155,6 +157,32 @@ inline std::string GenerateUUIDV4() {
   return ss.str();
 }
 
+/// Compress the input data using zstd compression
+inline std::string CompressZstd(const std::string &data) {
+  // Compress the data.
+  const size_t compressed_size = ZSTD_compressBound(data.size());
+  std::string compressed_data(compressed_size, '\0');
+  const size_t actual_compressed_size =
+      ZSTD_compress(compressed_data.data(), compressed_size, data.data(), data.size(), 1);
+  compressed_data.resize(actual_compressed_size);
```

Comment:
qq. Does resize do data copy?

Reply:
from the [documentation](https://en.cppreference.com/w/cpp/string/basic_string/resize) it's unclear (when new size is smaller than old size, which is most likely the case here). quick verification shows 
```
  std::string s(100, '\0');
  printf("%p\n", s.data());
  s.resize(20);
  printf("%p\n", s.data());

0x60b0000000f0
0x60b0000000f0
```